scripts/process_diagnostic_messages.rs

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env rust-script
+//!
+//! ```cargo
+//! [dependencies]
+//! serde = { version = "1.0", features = ["derive"] }
+//! serde_json = "1.0"
+//! pathdiff = "0.2.1"
+//! regex = "1.5.4"
+//! ```
+
+use regex::Regex;
+use serde::Deserialize;
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+enum DiagnosticCategory {
+    Error,
+    Message,
+    Suggestion,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+struct DiagnosticMessageSpec {
+    category: DiagnosticCategory,
+    code: u32,
+}
+
+fn main() {
+    let args = env::args().collect::<Vec<_>>();
+    if args.len() < 2 {
```

Comment:
Maybe `!= 2`?

Reply:
Ya that would make sense, the Typescript version uses `<` (it's actually checking `< 3` since it expects to be invoked like `node [script_name] [json_file_path]` - I think technically this could be a little fragile in that regard because you could invoke `rust-script` scripts either "directly" via the shebang like this or as `rust-script ...` (in which case the right number here would be `3`, not `2`))